### ISSUE TITLE
Pass HDF5 exceptions up through SWIG

### ISSUE THREAD

**Issue by Yurlungur** [eschnett/SimulationIO#70]
For a python API to be self-contained, it needs to be able to handle exceptions that are thrown. HDF5 passes a lot of these and they currently cause the python interpreter to segfault. We need to use the following technology:
http://www.swig.org/Doc1.1/HTML/Exceptions.html

**Comment by eschnett** [comment]
What features does Python require from this? Is this only about "don't crash, raise a Python exception instead", or is something more elaborate required?

It look as if adding ten lines or so to define a Python exception handler is all we need. Is this correct?

**Comment by Yurlungur** [comment]
Yep, all we need is an exception handler. In particular, catching the H5
exceptions and raising python IOErrors higher up is basically the
functionality we want.
